src/ansi.rs

```diff
@@ -339,6 +339,24 @@ impl<'a> From<(&'a str, &'a [usize], Attr)> for AnsiString<'a> {
     }
 }
 
+impl<'a> std::ops::Add for AnsiString<'a> {
```

Comment:
I'm not sure about the mutating part, is it really necessary ?

Reply:
Since `std::ops::Add` takes by value not reference I don't see the issue? This way we don't allocate if we don't need to.